Reduce size of test-runner args

If we specify all test methods that are to be ran in a big enough project, we will easily reach the space available for command arguments set by Linux. 

This limit can be adjusted, but even with 65MB available there is still an issue for thousands of methods.

Trace:
```
java.lang.RuntimeException: java.io.IOException: Cannot run program "java" (in directory "/home/andre/Repos/flacoco/."): error=7, Argument list too long
	at eu.stamp_project.testrunner.EntryPoint.runGivenCommandLine(EntryPoint.java:602)
	at eu.stamp_project.testrunner.EntryPoint.runCoveredTestResultPerTestMethods(EntryPoint.java:496)
	at fr.spoonlabs.flacoco.core.coverage.framework.JUnit4Strategy.execute(JUnit4Strategy.java:23)
	at fr.spoonlabs.flacoco.core.coverage.CoverageRunner.getCoverageMatrix(CoverageRunner.java:34)
	at fr.spoonlabs.flacoco.localization.spectrum.SpectrumRunner.computeCoverageMatrix(SpectrumRunner.java:37)
	at fr.spoonlabs.flacoco.localization.spectrum.SpectrumRunner.run(SpectrumRunner.java:22)
	at fr.spoonlabs.flacoco.api.Flacoco.runDefault(Flacoco.java:32)
	at fr.spoonlabs.flacoco.cli.FlacocoMain.call(FlacocoMain.java:105)
	at fr.spoonlabs.flacoco.cli.FlacocoMain.call(FlacocoMain.java:25)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2311)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at fr.spoonlabs.flacoco.cli.FlacocoMain.main(FlacocoMain.java:96)
Caused by: java.io.IOException: Cannot run program "java" (in directory "/home/andre/Repos/flacoco/."): error=7, Argument list too long
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
	at eu.stamp_project.testrunner.EntryPoint.runGivenCommandLine(EntryPoint.java:591)
	... 16 more
Caused by: java.io.IOException: error=7, Argument list too long
	at java.lang.UNIXProcess.forkAndExec(Native Method)
	at java.lang.UNIXProcess.<init>(UNIXProcess.java:247)
	at java.lang.ProcessImpl.start(ProcessImpl.java:134)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
	... 17 more
```

--- a/pom.xml
+++ b/pom.xml
@@ -88,7 +88,7 @@
         <dependency>
             <groupId>eu.stamp-project</groupId>
             <artifactId>test-runner</artifactId>
-            <version>4.8</version>
+            <version>4.9</version>
         </dependency>
 
         <dependency>

--- a/src/main/java/fr/spoonlabs/flacoco/core/coverage/framework/JUnit4Strategy.java
+++ b/src/main/java/fr/spoonlabs/flacoco/core/coverage/framework/JUnit4Strategy.java
@@ -13,8 +13,6 @@ public class JUnit4Strategy extends TestFrameworkStrategy {
 
 	private static final Logger logger = Logger.getLogger(JUnit4Strategy.class);
 
-	private static JUnit4Strategy instance;
-
 	public JUnit4Strategy(FlacocoConfig config) {
 		super(config);
 	}
@@ -29,7 +27,7 @@ public class JUnit4Strategy extends TestFrameworkStrategy {
 				config.getBinJavaDir(),
 				config.getBinTestDir(),
 				testContext.getTestMethods().stream().map(TestMethod::getFullyQualifiedClassName).distinct().toArray(String[]::new),
-				new String[0]
+				testContext.getTestMethods().stream().map(TestMethod::getFullyQualifiedMethodName).distinct().toArray(String[]::new)
 		);
 	}
 

--- a/src/main/java/fr/spoonlabs/flacoco/core/coverage/framework/JUnit5Strategy.java
+++ b/src/main/java/fr/spoonlabs/flacoco/core/coverage/framework/JUnit5Strategy.java
@@ -13,8 +13,6 @@ public class JUnit5Strategy extends TestFrameworkStrategy {
 
 	private static final Logger logger = Logger.getLogger(JUnit5Strategy.class);
 
-	private static JUnit5Strategy instance;
-
 	public JUnit5Strategy(FlacocoConfig config) {
 		super(config);
 	}
@@ -32,7 +30,7 @@ public class JUnit5Strategy extends TestFrameworkStrategy {
 				config.getBinJavaDir(),
 				config.getBinTestDir(),
 				testContext.getTestMethods().stream().map(TestMethod::getFullyQualifiedClassName).distinct().toArray(String[]::new),
-				new String[0]
+				testContext.getTestMethods().stream().map(TestMethod::getFullyQualifiedMethodName).distinct().toArray(String[]::new)
 		);
 	}
 

--- a/src/main/java/fr/spoonlabs/flacoco/core/coverage/framework/TestFrameworkStrategy.java
+++ b/src/main/java/fr/spoonlabs/flacoco/core/coverage/framework/TestFrameworkStrategy.java
@@ -29,6 +29,7 @@ public abstract class TestFrameworkStrategy {
 	 * Auxiliary method to setup test-runners default options
 	 */
 	protected void setupTestRunnerEntryPoint() {
+		EntryPoint.useOptionsFile = true;
 		EntryPoint.coverageDetail = ParserOptions.CoverageTransformerDetail.DETAIL_COMPRESSED;
 		EntryPoint.workingDirectory = new File(config.getWorkspace());
 		EntryPoint.verbose = config.isTestRunnerVerbose();

